### PR TITLE
Querying

### PR DESCRIPTION
**Pull request by Jay459**

### CODE REVIEW

File: src/controllers/teacher.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export default class TeacherController {
             }
 
         } catch (error) {
-            ctx.body = { message: error.message }
+            console.log(error)
```
Comment (priyankcommits) [inline src/controllers/teacher.ts]: why remove ctx.body ?

File: src/services/teacher.ts [hunk 2]
```diff
@@ -50,13 +50,32 @@ export class TeacherService {
         if (academicYearId === undefined) {
 
             // const teachers = await this.teacherRepository.find({ relations: ['subject'] });
+            // const teacherList = []
```
Comment (priyankcommits) [inline src/services/teacher.ts]: now if everything is working, lets remove all the previous commented out code.

File: src/controllers/teacher.ts (hunk 1 above)
Comment (Jay459) [reply src/controllers/teacher.ts]: i just want to check the error as it is not printing the whole error in postman so i have console logged the error just to debug the error

File: src/services/teacher.ts [hunk 3]
```diff
@@ -42,57 +33,86 @@ export class TeacherService {
...
-                    academicYear: academicYearId
-                }, relations: ['subject']
-            })
+            const pastTeachers = await this.teacherArchiveRepository.query('SELECT  "Teacher_Archive"."id" AS "Teacher_Archive_id", "Teacher_Archive"."name" AS "Teacher_Archive_name", "Teacher_Archive"."gender" AS "Teacher_Archive_gender", "Teacher_Archive__subject"."id" AS "Teacher_Archive__subject_id", "Teacher_Archive__subject"."name" AS "Teacher_Archive__subject_name" FROM "Teacher_Archive" "Teacher_Archive" LEFT JOIN "Subject_Archive" "Teacher_Archive__subject" ON "Teacher_Archive__subject"."teacherId"="Teacher_Archive"."id"')
 
             if (pastTeachers.length <= 0) {
                 throw new Error("There are no teachers in this Year");
             } else {
-                return pastTeachers;
+                let changedTeachers = [];
```
Comment (priyankcommits) [inline src/services/teacher.ts]: all this code is same as above, refactor to a common function and call that function from both places

**Pull request merged**